encrypt.py

```diff
@@ -18,91 +22,134 @@
-    plaintext=plaintext_of_sym_key, conditions=eth_balance_condition
+@click.option(
+    "--coordinator-provider-uri", type=str, help="URI of the coordinator provider", required=True
 )
-
-tmk = {
-    "bulk_ciphertext": base64.b64encode(bytes(bulk_ciphertext)).decode(),  # Encrypted Tony
-    "encrypted_sym_key": bytes(ciphertext_of_sym_key).hex(),
-    "conditions": eth_balance_condition,
-    "filename": "manzana.mp3",
-}
-
-################
-# Sanity check #
-################
-
-f = Fernet(plaintext_of_sym_key)
-hopefully_tony = f.decrypt(bulk_ciphertext)
-assert hopefully_tony == definitely_tony
-
-##################
-
-tmk_json = json.dumps(tmk)
-
-filename = "tony.tmk"
-with open(filename, "w") as file:
-    data = tmk_json
-    file.write(data)
-    print(f"Wrote {len(data)} bytes to {filename}")
+@click.option(
+    "--coordinator-network",
+    default="mumbai",
+    help="Network for the coordinator",
+    show_default=True,
+    type=click.Choice(["mumbai", "rinkeby", "mainnet", "goerli", "ropsten", "kovan"]),
+)
+@click.option("--chain", type=int, help="Ethereum chain ID", default=80001, show_default=True)
+@click.option(
+    "--eth-address",
+    type=str,
+    help="Ethereum address for balance check",
+    default="0x210eeAC07542F815ebB6FD6689637D8cA2689392",
+    show_default=True,
+)
+@click.option(
+    "--eth-minimum-balance",
+    type=float,
+    help="Ethereum minimum balance condition",
+    default=0,
+    show_default=True,
+)
+@click.option("--output-file", type=str, default="tony.tmk", help="Output file for encrypted data")
+def main(
+    input_file: str,
+    ritual_id: int,
+    coordinator_provider_uri: str,
+    coordinator_network: str,
+    chain: int,
+    eth_address: str,
+    eth_minimum_balance: float,
+    output_file: str,
+):
+    file_path = Path(input_file)
+
+    with open(file_path, "rb") as f:
+        cleartext = f.read()
+
+    plaintext_of_sym_key = keygen()
+
+    secret_hash = keccak(plaintext_of_sym_key)
+    bulk_ciphertext = encapsulate(plaintext_of_sym_key, cleartext)
+
+    print("--------- Threshold Encryption ---------")
+
+    coordinator_agent = CoordinatorAgent(
+        provider_uri=coordinator_provider_uri,
+        registry=InMemoryContractRegistry.from_latest_publication(network=coordinator_network),
+    )
+    ritual = coordinator_agent.get_ritual(ritual_id)
+    enrico = Enrico(encrypting_key=DkgPublicKey.from_bytes(bytes(ritual.public_key)))
+
+    print(
+        f"Fetched DKG public key {bytes(enrico.policy_pubkey).hex()} "  # type: ignore
+        f"for ritual #{ritual_id} "
+        f"from Coordinator {coordinator_agent.contract.address}"
+    )
+
+    eth_balance_condition: Lingo = {
+        "version": ConditionLingo.VERSION,
+        "condition": {
+            "chain": chain,
+            "method": "eth_getBalance",
+            "parameters": [eth_address, "latest"],
+            "returnValueTest": {"comparator": ">=", "value": eth_minimum_balance},
```

Comment:
I changed the comparator and added an `eth_minimim_balance` parameter, because in the final incrypt we want to set this to 10 ETH and not 0, right?